src/compiler/checker.ts

```diff
@@ -135,6 +135,7 @@ namespace ts {
             getDeclaredTypeOfSymbol,
             getPropertiesOfType,
             getPropertyOfType: (type, name) => getPropertyOfType(type, escapeLeadingUnderscores(name)),
+            getPropertyForPrivateName,

```

Comment:
where is this used outside the checker? I couldn't find it, in which case it doesn't need to be exported.

Reply:
Will TypeScript API consumers need this to look up a private named property? Also, if it remains should it be `getPropertyForPrivateIdentifier`?

Reply:
This function is exported publicly because `getPropertyOfType` isn't really sufficient for getting the type of a private identifier property (since that function takes the escaped name of the property and checks if a symbol with that name exists - for private identifiers, the escaped name of the property isn't unique, you also need the class that the private identifier is on to produce the fully qualified symbol name). Without this function, consumers of the API won't really have a (non-hacky) mechanism to get the symbol for a private identifier property.

Should we name it something like `getPrivatelyIdentifiedPropertyOfType` (or more simply `getPrivatePropertyOfType` if it's not confusing with design-time privacy)?

Reply:
we renamed to `getPropertyForPrivateIdentifier`. It is currently both exposed and used internally.